android/src/main/java/org/reactnative/camera/tasks/ResolveTakenPictureAsyncTask.java

```diff
@@ -108,12 +115,19 @@ protected WritableMap doInBackground(Void... voids) {
                 int orientation = exifInterface.getAttributeInt(ExifInterface.TAG_ORIENTATION,
                         ExifInterface.ORIENTATION_UNDEFINED);
 
+                int cameraOrientation = getCameraOrientation();
```

Comment:
You are getting the camera orientation after the picture has been taken in an async task. The phone might have been rotated by now to another position. Have you tried what happens in such case? You will probably be using a rotation that's different than what the picture was taken with.  You may need to do this on the picture captured events of each camera implementation.

Reply:
You can probably already get the camera orientation from each Camera implementation from these lines. The implementation will already handle getting the orientation from the right camera.

https://github.com/react-native-community/react-native-camera/blob/master/android/src/main/java/com/google/android/cameraview/Camera1.java#L768
https://github.com/react-native-community/react-native-camera/blob/master/android/src/main/java/com/google/android/cameraview/Camera2.java#L693

So rather than getting the camera orientation after the picture has been taken and the image is getting processed, you may be able to get it right before the capture event from those lines, and pass it down to the async task. What do you think? That would ensure you always get the orientation from the right camera, at the right time.

Reply:
I pushed new commit. Please, check this out